lib/kredis/types/unique_list.rb

```diff
@@ -1,29 +1,84 @@
 # You'd normally call this a set, but Redis already has another data type for that
-class Kredis::Types::UniqueList < Kredis::Types::List
-  proxying :multi, :ltrim, :exists?
+class Kredis::Types::UniqueList < Kredis::Types::Proxying
+  proxying :multi, :zrange, :zrem, :zadd, :zremrangebyrank, :exists?, :del
 
   attr_accessor :typed, :limit
 
-  def prepend(elements)
-    elements = Array(elements).uniq
-    return if elements.empty?
+  def elements
+    auto_fallback(:elements) do
+      strings_to_types(zrange(0, -1) || [], typed)
+    end
+  end
+  alias to_a elements
 
-    multi do
-      remove elements
-      super
-      ltrim 0, (limit - 1) if limit
+  def remove(*elements)
+    auto_migrate do
+      zrem(types_to_strings(elements, typed))
     end
   end
 
-  def append(elements)
-    elements = Array(elements).uniq
-    return if elements.empty?
+  def prepend(elements)
+    auto_migrate do
+      insert(elements, prepending: true)
+    end
+  end
 
-    multi do
-      remove elements
-      super
-      ltrim -limit, -1 if limit
+  def append(elements)
+    auto_migrate do
+      insert(elements)
     end
   end
   alias << append
+
+  private
+    def insert(elements, prepending: false)
+      elements = Array(elements)
+      return if elements.empty?
+
+      elements_with_scores = types_to_strings(elements, typed).map do |element|
+        [ current_nanoseconds(negative: prepending), element ]
+      end
+
+      multi do
+        zadd(elements_with_scores)
```

Comment:
Consider [LT / GT options](https://redis.io/commands/ZADD#zadd-options) for ordering safety, avoiding problems with clock skew.